src/reactionsystem.jl

```diff
@@ -1221,7 +1222,34 @@ function set_default_metadata(rs::ReactionSystem;  default_reaction_metadata = [
     updated_equations = map(eqtransform, get_eqs(rs))
     @set! rs.eqs = updated_equations
     @set! rs.rxs = Reaction[rx for rx in updated_equations if rx isa Reaction]
-    
+
+    # Special routine to handle `Reaction` metadata that can contain new symbolic variables.
+    # Currently, `noise_scaling` is the only relevant metadata supported this way.
+    drm_dict = Dict(default_reaction_metadata)
+    if haskey(drm_dict, :noise_scaling)
+        # Finds parameters, species, and variables in the noise scaling term.
+        ns_expr = drm_dict[:noise_scaling]
+        ns_syms = [Symbolics.unwrap(sym) for sym in get_variables(ns_expr)]
```

Comment:
`unwrapping` should be handled in the `Reaction` constructor. We shouldn't be storing things that aren't unwrapped as we don't want to have to think about this internally later on (at least every internal Catalyst function I've written assumes its input is unwrapped as this is the MTK convention).

Reply:
Right now we just use the same convention for `Reaction` metadata as MTK/Symbolics does for symbolic variables (i.e. metadata is stored as they are provided by the user, if the user sets a metadata as `Num` it remain as a `Num`).

If you want we can add a specific routine in the `Reaction` constructor that checks if there is any `noise_scaling` metadata, and if it is `Num` unwraps it at that stage.